fluss-server/src/test/java/org/apache/fluss/server/RpcServiceBaseTest.java

```diff
@@ -0,0 +1,758 @@
+import org.apache.fluss.server.metadata.MetadataFunctionProvider;
+import org.apache.fluss.server.metadata.PartitionMetadata;
+import org.apache.fluss.server.metadata.ServerMetadataCache;
+import org.apache.fluss.server.metadata.TableMetadata;
+import org.apache.fluss.server.zk.ZooKeeperClient;
+import org.apache.fluss.server.zk.data.BucketAssignment;
+import org.apache.fluss.server.zk.data.LeaderAndIsr;
+import org.apache.fluss.server.zk.data.PartitionAssignment;
+import org.apache.fluss.server.zk.data.TableAssignment;
+
+import org.assertj.core.util.Lists;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.lang.reflect.Field;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutionException;
+
+import static org.apache.fluss.record.TestData.DATA1_PARTITIONED_TABLE_DESCRIPTOR;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+/** Unit test for {@link RpcServiceBase} async metadata methods. */
+class RpcServiceBaseTest {
+
+    private ZooKeeperClient mockZkClient;
+    private TablePath testTablePath;
+    private PhysicalTablePath testPartitionPath;
+    private long testTableId;
+    private long testPartitionId;
+    private List<BucketMetadata> testBucketMetadata;
+
+    @BeforeEach
+    void beforeEach() {
+        mockZkClient = mock(ZooKeeperClient.class);
+
+        testTablePath = TablePath.of("testDb", "testTable");
+        testPartitionPath = PhysicalTablePath.of(testTablePath, "partition1");
+        testTableId = 1L;
+        testPartitionId = 100L;
+        testBucketMetadata =
+                Lists.newArrayList(
+                        new BucketMetadata(0, 1, 1, Arrays.asList(1, 2, 3)),
+                        new BucketMetadata(1, 2, 2, Arrays.asList(4, 5, 6)));
+    }
+
+    private Session createTestSession() throws java.net.UnknownHostException {
+        return new Session(
+                (short) 1,
+                "default",
+                false,
+                java.net.InetAddress.getByName("127.0.0.1"),
+                FlussPrincipal.ANY);
+    }
+
+    // A test implementation of RpcServiceBase to access the protected processMetadataRequest method
+    private static class TestRpcServiceBase extends RpcServiceBase {
+        public TestRpcServiceBase() {
+            super(null, null, null, null, null);
+        }
+
+        @Override
+        public void processMetadataRequest(
```

Comment:
Why override this method? It seems we didn't do anything except calling parent method?